### PR TITLE
register federation failure metrics

### PR DESCRIPTION
**Pull request by juliantaylor**
Closes gh-7080

### CODE REVIEW

**Comment by brian-brazil** [comment]
The tests are failing.

File: web/federate.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ var (
 	})
 )
 
+func registerFederationMetrics(r prometheus.Registerer) {
```
Comment (roidelapluie) [inline web/federate.go:47]: Why do we need a function?

Comment (juliantaylor) [reply web/federate.go:47]: so the metric registration is in the same file as the metrics themselves so it is less likely a register is forgotten when adding metrics

do you prefer it differently?

**Comment by juliantaylor** [comment]
test failures are fixed

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**